File: src/main/java/org/flossware/civilization/model/Scenario.java
package org.flossware.civilization.model;

import java.util.List;

/**
 * Complete scenario configuration.
 */
public record Scenario(
    String scenarioId,
    String name,
    String description,
    int startYear,
    int endYear,
    CivilizationState initialState,
    List<Technology> techTree,
    WorldConstraints worldConstraints,
    SimulationRules simulationRules
) {
    public Scenario {
        if (startYear >= endYear) {
            throw new IllegalArgumentException("Start year must be before end year");
        }
        techTree = List.copyOf(techTree);
    }
}

File: mobile/shared/src/commonMain/kotlin/org/flossware/civilization/model/Scenario.kt
package org.flossware.civilization.model

import kotlinx.serialization.Serializable

@Serializable
data class Scenario(
    val scenarioId: String,
    val name: String,
    val description: String,
    val startYear: Int,
    val endYear: Int,
    val initialState: CivilizationState,
    val techTree: List<Technology>,
    val worldConstraints: WorldConstraints,
    val simulationRules: SimulationRules
) {
    init {
        require(startYear < endYear) { "Start year must be before end year" }
    }
}
